hooks/src/index.js

```diff
@@ -346,9 +352,17 @@ function invokeEffect(hook) {
 /**
  * @param {any[]} oldArgs
  * @param {any[]} newArgs
+ * @param {import('./internal').Component} currentComponent
  */
-function argsChanged(oldArgs, newArgs) {
-	return !oldArgs || oldArgs.length !== newArgs.length || newArgs.some((arg, index) => arg !== oldArgs[index]);
+function argsChanged(oldArgs, newArgs, currentComponent) {
+	if (options._argsChanged) {
```

Comment:
Does it make any difference if we extend the `_hook` option to handle this case? Perhaps something like `getHookState(currentIndex++, 7, args)` (we probably only need to pass `args` if the hook has an `args` param)

Looking at it now, should we invoke that `_hook` option after we have created/retrieved the hook state so we can pass it to the hook? Sounds like that would make the `_hook` option a lot more useful/powerful.

Will make the debug code more complex but if it saves bytes in product code, maybe that's worth it

Reply:
Just did that with the latest commit, but it turns out to be bigger than expected. It's 15B larger compared to the additional option hook. Another argument in favor of keeping `options._hook` as is, is that it's a breaking change, even though it's internals. Both devtools and prefresh are built on it.